react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js

```diff
@@ -8,16 +8,26 @@ const initialState = {
 
 function reducer(state = initialState, action) {
   switch (action.type) {
-    case actions.GET_BOOKS: // TODO to implement the logic
-      return { ...state };
-    case actions.ADD_TO_CART: // TODO to implement the logic
-      return { ...state };
-    case actions.ADD_ITEM: // TODO to implement the logic
-      return { ...state };
-    case actions.REMOVE_ITEM: // TODO to implement the logic
-      return { ...state };
-    case actions.SEARCH_ITEM: // TODO to implement the logic
-      return { ...state };
+    case actions.GET_BOOKS:
+      return { ...state, books: action.payload, originalData: action.payload };
```

Comment:
Why are you assigning the payload to two different state properties here?

Also, we usually use `state.merge()` to keep the state object `Immutable`

Reply:
because i have to use originalData to filter when i try to search by name